src/Raven.Server/Rachis/Leader.cs

```diff
@@ -1105,7 +1107,9 @@ private void ValidateUsableReturnType(object result)
             if (result is BlittableJsonReaderObject || result is BlittableJsonReaderArray)
                 throw new RachisApplyException("You cannot return a blittable here, it is bound to the context of the state machine, and cannot leak outside");
 
-            if (TypeConverter.IsSupportedType(result) == false)
+            var jsEngineType = JavaScriptEngineType.Jint; // TODO [shlomo] check if this is correct: for IsSupportedType there should not be a difference
+                            
+            if (TypeConverterBase.IsSupportedType(jsEngineType, result) == false)
```

Comment:
Why does `IsSupportedType` need a JS engine?
Because of this: https://github.com/ravendb/ravendb/blob/f19b7f4f4d32dd4a1236992177b917e001bb5b6c/src/Raven.Server/Utils/TypeConverter.cs#L270

But we have different types here, we can detect and use the right one, instead.

Reply:
I have merged Jint and V8 versions and simplified TypeConverter class, turned it back to be static, and eliminated jsEngineType arguments in ToBlittableSupportedType and IsSupportedType